### DISCUSSION TITLE
v4.5.1 (Beta)

### DISCUSSION THREAD

**Discussion by osy** [utmapp/UTM#6241] [Releases]
## Highlights
* **UTM Remote server for macOS** On macOS 13+, you can enable UTM Server from the new option on the home screen or from Window -> UTM Server. Once enabled, you can stream QEMU backend VMs to supported clients. The preferences page includes additional options including auto-starting the server and allowing external connections so it can be used outside of the local network. New documentation pages will be added in the future.
* **UTM Remote client for iOS and visionOS** When UTM Server is enabled on macOS, you can connect to it with the new UTM Remote client which uses the same frontend as UTM for iOS/visionOS but without any of the QEMU backend. TestFlight will be available as soon as it is approved and the plans are to release it as a free app in the App Store.

## Notes
* (macOS) The automation URI scheme has been removed due to potential security issues if the user clicks a malicious link that sends an input to the VM. Most of the URI scheme functionality has moved to [the scripting interface](https://docs.getutm.app/scripting/scripting/).

## Changes (v4.5.1)
* Fixed an issue with file locking resulting in VMs refusing to boot (#5757, #5830)
* Fixed incorrect VM size calculation (#6166)
* Show confirmation popup for VM downloads (#6156)
* Removed automation URI scheme due to potential security issues (#6155)
* Fixed display of newly selected custom icon (#6137)
* Wizard: allow completely deleting RAM and storage size (#5885)
* Scripting: Fixed file and process commands not working due to incorrect object life cycle (#5963)
* Localization: Updated Japanese (thanks @MMP0)
* Localization: Updated Polish (thanks @mavethee)
* Localization: Updated Chinese (Simplified + Traditional) (thanks @changanmoon)
* (iOS) Fixed zooming with pinch (#6179)
* (macOS) Fixed USB icon always disabled (#6222)
* (macOS) Fixed a crash when removing a device while a text field is highlighted (#5901)
* (macOS) Fixed error message when double-clicking on a headless VM which has already been started (#5972)
* (macOS) AVF: New display option to disable dynamic resolution in macOS 14+ VMs (#5873)
* (macOS) Remove "VM display size is fixed" global setting because it was confusing and does not do the right thing
* (macOS) Fixed incorrect display scaling when host screen resolution is smaller than VM display size (#6214)

## Changes (v4.5.0)
* Updated ANGLE to latest Safari version
* Fixed a crash due to screenshot being saved while the image was being destroyed (#4009)
* Fixed a memory leak caused by a retain cycle while observing changes in the VM state
* Localization: Added Italian (thanks @garamb1)
* Localization: Updated Japanese (thanks @MMP0)
* Localization: Updated Polish (thanks @mavethee)
* Localization: Updated Chinese (Simplified + Traditional) (thanks @changanmoon)
* (macOS) Add UTM Server feature
* (iOS) Fixed the icon name for UTM SE erroneously set to UTM in translation (#5968)
* (iOS) Disable logging of keyboard keys when Debug Logs are enabled. They aren't helpful for any issue debugging and poses a privacy issue.
* (iOS) Reworked automatic resolution changing when SPICE tools are installed
* (iOS) Fix crash due to race when re-sizing while a view is being destroyed
* (visionOS) Move toolbar to top of window and support hiding
* (visionOS) Make VM windows less rounded
* (visionOS) Integrate new keyboard view

## Issues
Please check the full list on [Github](https://github.com/utmapp/UTM/issues) and help report any bug you find that is not listed.

## Installation

* [iOS](https://docs.getutm.app/installation/ios/)
* [macOS](https://docs.getutm.app/installation/macos/)

| File | Description | Installation | JIT | Hypervisor | USB |
|------|------------|--------------|-----|-----------|-----|
| UTM.deb | Jailbroken iOS version | Open in Cydia, dpkg, or Sileo | Yes | Yes(1) | Yes |
| UTM.dmg | macOS version | Mounting and copying UTM.app to /Applications | Yes | Yes | Yes |
| UTM.ipa | Non-jailbroken iOS version (sideloading) | AltStore, etc (see guide) | Yes(2) | No | No |
| UTM-HV.ipa | Non-jailbroken iOS version (TrollStore) | TrollStore | Yes | Yes(1) | Yes |
| UTM-SE.ipa | Non-jailbroken iOS version (sideloading) | AltStore, enterprise signing, etc | No | No | No |
| UTM-Remote.ipa | Remote client | Any | No | No | No |

1. Hypervisor on iOS requires an M1 iPad.
2. Enabling JIT may require a separate JIT enabler such as [Jitterbug][2] or Jitstreamer.

[1]: https://getutm.app/install/
[2]: https://github.com/osy/Jitterbug

<hr /><em>This discussion was created from the release <a href='https://github.com/utmapp/UTM/releases/tag/v4.5.1'>v4.5.1 (Beta)</a>.</em>

**Comment by romlu2024** [comment]
Hi @osy, the UTM.dmg asset is missing for version 4.5.1 (Beta) ;-)

**Reply by osy** [reply]
Be patient, every time someone comes in < 1 hour and asks where the DMG is when it's still being processed.

**Reply by osy** [reply]
Okay it should be up now

**Reply by romlu2024** [reply]
Cool, thank you :-)

**Comment by zjysun** [comment]
Thank you very much for resolving the resolution issue on UTM for iOS devices; my testing on version 4.5.0 beta has shown that the resolution issues on Windows systems have indeed been resolved. However, I've encountered persistent resolution problems on Ubuntu 22.04 systems. And in the latest 4.5.1 beta version, this problem still exists. My current workaround is to switch the UTM app to the background and then re-enter, at which point the resolution will automatically adjust once and return to normal, as demonstrated in the attached video. Additionally, although the resolution issues on Windows are now fixed, I've noticed that since version 4.2.5 of UTM, there appears to be a persistent issue on iOS systems related to hiding the bottom black bar (Home Indicator), which does not automatically hide. I have attached videos regarding these two issues. I hope these issues can receive your attention and be addressed. Many thanks to the development team for your efforts!

https://github.com/utmapp/UTM/assets/8925908/087d7882-74b3-4ce2-97e1-25e52b7b8a29
